File: apps/src/chat/frontend/js/packets/UserPacket.js
export default class UserPacket {

    constructor(
        username
    ){
        this.username = username;
    }

    toData(){
        return {
            username: this.username
        }
    }

    static fromData(data){
        return new UserPacket(data["username"]);
    }

    getUsername(){
        return this.username;
    }
}

File: apps/src/chat/backend/packets/http/user_packet.py
from ..packet import Packet

class UserPacket(Packet):

    def __init__(self, username: str):
        self._username = username

    def get_type(self):
        return "user"

    def to_data(self):
        return {
            "type": self.get_type(),
            "username": self.get_username()
        }

    @staticmethod
    def from_data(data):
        return UserPacket(data["username"])

    def get_username(self) -> str:
        return self._username